compiler/rustc_arena/src/lib.rs

```diff
@@ -216,7 +217,7 @@ impl<T> TypedArena<T> {
 
     #[inline]
     fn can_allocate(&self, additional: usize) -> bool {
-        let available_bytes = self.end.get() as usize - self.ptr.get() as usize;
+        let available_bytes = self.end.get().addr() - self.ptr.get().addr();
```

Comment:
I wonder if `lhs.addr().wrapping_sub(rhs.addr())` is a common enough use case to deserve its own method.

Reply:
I believe this should just use `offset_from`.

Reply:
Oh, hmm, that method does exist. No wrapping_offset_from shough.

Reply:
Yeah, wrapping_offset_from got remove since it's just subtracting the integers...

Reply:
Oh blech. When I looked for `offset_from` I hit `align_offset` and wire-crossed to "oh right you can't do this operation because its guarantees are completely wild and unusable".

Reply:
> Yeah, wrapping_offset_from got removed since it's just subtracting the integers...

Well, I guess we got our use case now. :p

Reply:
wait why would this need to use wrapping?

Reply:
Not in this case, but... actually, good point, there's not really any case besides with_addr that would need it, at least that I can think of.